src/panels/config/entities/ha-config-entities.ts

```diff
@@ -196,10 +196,23 @@ export class HaConfigEntities extends LitElement {
     }
   );
 
+  public hassSubscribe(): UnsubscribeFunc[] {
+    return [
+      subscribeEntityRegistry(this.hass.connection!, (entities) => {
+        this._entities = entities;
+      }),
+    ];
+  }
+
   public disconnectedCallback() {
     super.disconnectedCallback();
-    if (this._unsubEntities) {
-      this._unsubEntities();
+    const dialog = document
```

Comment:
Eeeh, this is not nice. Can't we return a function from `showXDialog` that allows closing it ?

Reply:
Yeah this is ugly... that sounds good will try 👍

Reply:
You can put this exact logic in `showXDialog`, but that should be the only place that knows about the HTML tag name of the dialog element.